test/e2e/targets/azureeventhubs/main.go

```diff
@@ -0,0 +1,278 @@
+  - AZURE_CLIENT_SECRET - The Azure ServicePrincipal Client Secret
+  - AZURE_REGION - Define the Azure region to run the test (default westus2)
+
+  These will be done by the e2e test:
+  - Create an Azure Resource Group, EventHubs Namespace, and EventHub
+  - Send an event to the Azure EventHub via the AzureEventHub Target and verify the event was sent
+*/
+
+var _ = FDescribe("Azure EventHubs target", func() {
+	ctx := context.Background()
+	subscriptionID := os.Getenv("AZURE_SUBSCRIPTION_ID")
+	region := os.Getenv("AZURE_REGION")
+
+	if region == "" {
+		region = "westus2"
+	}
+
+	f := framework.New("azureeventhubstarget")
+
+	var ns string
+	var tgtURL *url.URL
+	var tgtClient dynamic.ResourceInterface
+
+	var rg armresources.ResourceGroup
+	var hub *eventhubs.Hub
+
+	BeforeEach(func() {
+		ns = f.UniqueName
+
+		gvr := targetAPIVersion.WithResource("azureeventhubstargets")
+		tgtClient = f.DynamicClient.Resource(gvr).Namespace(ns)
+	})
+
+	Context("a target is deployed", func() {
+		BeforeEach(func() {
+			By("creating an azure resource group", func() {
+				rg = azure.CreateResourceGroup(ctx, subscriptionID, ns, region)
+				DeferCleanup(func() {
+					By("deleting the azure resource group", func() {
+						_ = azure.DeleteResourceGroup(ctx, subscriptionID, *rg.Name)
+					})
+				})
+			})
+
+			By("creating an azure eventhub", func() {
+				hub = azure.CreateEventHubComponents(ctx, subscriptionID, ns, region, *rg.Name)
+			})
+		})
+
+		When("the spec contains default settings", func() {
+			var event *cloudevents.Event
+
+			BeforeEach(func() {
+				By("creating an AzureEventHubTarget object", func() {
+					tgt, err := createTarget(tgtClient, ns, "test-",
+						withServicePrincipal(),
+						withEventHubID(subscriptionID, *rg.Name, ns, ns))
+
+					Expect(err).ToNot(HaveOccurred())
+
+					tgt = ducktypes.WaitUntilReady(f.DynamicClient, tgt)
+					tgtURL = ducktypes.Address(tgt)
+					Expect(tgtURL).ToNot(BeNil())
+				})
+			})
+
+			It("receives an event on the Event Bus", func() {
+				var payload []byte
+				var partitionIDs []string
+				var eventHandler eventhubs.Handler
+				eventReceivedChannel := make(chan bool)
+				var evCtx context.Context       // Used to set a timeout for reading events
+				var evCancel context.CancelFunc // Used to cancel the reading context
+
+				By("retrieving eventhub partition details", func() {
+					info, err := hub.GetRuntimeInformation(ctx)
+					Expect(err).NotTo(HaveOccurred())
+
+					partitionIDs = info.PartitionIDs
+					Expect(len(partitionIDs) > 0).To(BeTrue())
+				})
+
+				By("creating a handler to verify the received event", func() {
+					eventHandler = func(ctx context.Context, ev *eventhubs.Event) error {
+						defer GinkgoRecover() // To circumvent being called from inside a goroutine
+
+						payload = ev.Data
+						Expect(len(payload) > 0).To(BeTrue())
+
+						// NOTE: The payload will be a stringified version of the cloudevent
```

Comment:
Could be interesting given the `cloudenvents.Event.String()` function serializes the string more into a human readable context instead of something that can be machine parseable. Would rather tackle this in a follow-up PR that corrects how the full event is sent to the Azure EventHub.

Reply:
I was more thinking about about using `json.Unmarshal(<pointer to cloudevent.Event>)`

Reply:
Please ignore the above comment, I see what you mean now.